src/signalrclient/hub_connection.cpp

```diff
@@ -17,17 +17,6 @@ namespace signalr
         : m_pImpl(hub_connection_impl::create(url, trace_level, log_writer, http_client, websocket_factory))
     {}
 
-    hub_connection::hub_connection(hub_connection&& rhs) noexcept
```

Comment:
Weren't we going to turn hub_connection_impl into hub_connection? Otherwise, we're essentially returning a share_ptr<shared_ptr<real_hub_connection>>. While not necessarily wrong, I don't see the point.

Reply:
I gave a little explanation in the initial PR comment

Reply:
> We talked about replacing the hub_connection_impl and making hub_connection contain all the logic. However, that would require including a lot of internal types in the public header and all sorts of ugliness.

There has to be *some way* to get rid of the redundant share_ptr without exposing internal types, no?